src/Compilers/Core/Portable/Collections/UnionCollection.cs

```diff
@@ -42,7 +42,24 @@ public static ICollection<T> Create(ICollection<T> coll1, ICollection<T> coll2)
                 return coll1;
             }
 
-            return new UnionCollection<T>(ImmutableArray.Create(coll1, coll2));
+            return new UnionCollection<T>(Flatten(ImmutableArray.Create(coll1, coll2)));
+        }
+
+        public static ICollection<T> Create(ImmutableArray<ICollection<T>> collections)
+        {
+            Debug.Assert(collections.All(c => c.IsReadOnly));
+
+            switch (collections.Length)
```

Comment:
Indeed, a switch expression here would be nice :)
https://github.com/CyrusNajmabadi/roslyn/pull/8/files#r367126784